src/lock_api.rs

```diff
@@ -0,0 +1,123 @@
+//! Mock implementation of the `lock_api` and `parking_lot` crates.
+//!
+//! _These types are only available if loom is built with the `"lock_api"`
+//! feature_
+
+use crate::rt;
+use lock_api_ as lock_api;
+use once_cell::sync::OnceCell;
+
+/// Mock implementation of `lock_api::RawMutex`
+#[allow(missing_debug_implementations)]
+pub struct RawMutex {
+    object: OnceCell<rt::Mutex>,
+}
+
+impl RawMutex {
+    // Unfortunately, we're required to have a `const INIT` in order to
+    // implement `lock_api::RawMutex`, so we need to lazily create the actual
+    // `rt::Mutex`.
+    fn object(&self) -> &rt::Mutex {
+        self.object.get_or_init(|| rt::Mutex::new(true))
+    }
```

Comment:
I couldn't think of a cleaner way to satisfy this requirement, so I threw in a OnceCell.